pkg/controller/backup/backup_controller.go

```diff
@@ -0,0 +1,173 @@
+	corev1 "k8s.io/api/core/v1"
+	resource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	ControllerName = "backup-controller"
+	DataMountPath  = "/opt/infinispan/backups"
+)
+
+var ctx = context.Background()
+
+// reconcileBackup reconciles a Backup object
+type reconcileBackup struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client.Client
+}
+
+type backupResource struct {
+	instance *v2.Backup
+	client   client.Client
+	scheme   *runtime.Scheme
+}
+
+func Add(mgr manager.Manager) error {
+	return zero.CreateController(ControllerName, &reconcileBackup{mgr.GetClient()}, mgr)
+}
+
+func (r *reconcileBackup) ResourceInstance(key client.ObjectKey, ctrl *zero.Controller) (zero.Resource, error) {
+	instance := &v2.Backup{}
+	if err := ctrl.Get(ctx, key, instance); err != nil {
+		return nil, err
+	}
+
+	instance.Spec.ApplyDefaults()
+	return &backupResource{
+		instance: instance,
+		client:   r.Client,
+		scheme:   ctrl.Scheme,
+	}, nil
+}
+
+func (r *reconcileBackup) Type() runtime.Object {
+	return &v2.Backup{}
+}
+
+func (r *backupResource) AsMeta() metav1.Object {
+	return r.instance
+}
+
+func (r *backupResource) Cluster() string {
+	return r.instance.Spec.Cluster
+}
+
+func (r *backupResource) Phase() zero.Phase {
+	return zero.Phase(string(r.instance.Status.Phase))
+}
+
+func (r *backupResource) UpdatePhase(phase zero.Phase) error {
+	instance := r.instance
+	instance.Status.Phase = v2.BackupPhase(string(phase))
+	return r.updateStatus()
+}
+
+func (r *backupResource) Init() (*zero.Spec, error) {
+
+	// TODO add labels
+	pvc := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.instance.Name,
+			Namespace: r.instance.Namespace,
+		},
+	}
+
+	volumeSpec := r.instance.Spec.Volume
+	var storage resource.Quantity
+	if volumeSpec.Storage == nil {
+		// TODO calculate based upon number of Pods in cluster
+		// ISPN- Utilise backup size estimate
+		storage = constants.DefaultPVSize
+	} else {
+		storage = resource.MustParse(*volumeSpec.Storage)
+	}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, r.client, pvc, func() error {
+		pvc.Spec = corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteOnce,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: storage,
+				},
+			},
+			StorageClassName: volumeSpec.StorageClassName,
+		}
+		return controllerutil.SetControllerReference(r.instance, pvc, r.scheme)
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create pvc: %w", err)
+	}
+
+	pvcName := r.instance.Name
```

Comment:
Why you need to store PVC name as it always equals to CR name?

Reply:
I thought storing it in the Status would make it more explicit for users who might not be aware of this. For example if they do `kubectl describe Backup`

Reply:
Yes, sure. But in this case, the name should be FQN for PVC that is `persistentVolumeClaim`